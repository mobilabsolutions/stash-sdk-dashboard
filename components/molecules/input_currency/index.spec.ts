import { testRender, deepRender } from '../../../test_utils'

import InputCurrency from './index'
import { fireEvent } from 'react-testing-library'

it('Input Should Render', () =>
  testRender(InputCurrency, {
    field: { name: 'test', value: 12345 },
    form: { touched: { name: false }, errors: {} },
    currencyId: 'EUR'
  }))

it('Input Should fire onChange event with numeric value', () => {
  const onChange = jest.fn()
  const { getByPlaceholderText } = deepRender(InputCurrency, {
    field: { name: 'test', value: 12345, onChange },
    form: { touched: { name: false }, errors: {} },
    placeholder: 'test',
    currencyId: 'EUR'
  })
  const InputEl = getByPlaceholderText('test')
  fireEvent.change(InputEl, { target: { value: 'â‚¬112,345.00' } })
  expect(onChange).toBeCalledWith(
    expect.objectContaining({
      target: expect.objectContaining({
        value: 112345.0
      })
    })
  )
})
