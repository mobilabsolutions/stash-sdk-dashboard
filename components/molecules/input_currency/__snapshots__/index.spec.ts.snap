// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should Render 1`] = `
<div
  className="field_wrapper-n32k2e-0 irXinp"
  onClick={[Function]}
  style={Object {}}
>
  <div
    style={Object {}}
  />
  <div
    className="input__InputErrorWrapper-sc-1dz7i8-0 cAUOWl"
  >
    <div
      className="input_wrapper-sc-15hmhfy-0 duFQAu"
      style={Object {}}
    >
      <input
        autoFocus={false}
        className="input_currency__HtmlInput-obstnl-0 cVRLQV"
        disabled={false}
        name="test"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        placeholder=""
        type="text"
        value="â‚¬12,345.00"
      />
    </div>
  </div>
</div>
`;
